applications/StructuralMechanicsApplication/custom_elements/prestress_membrane_element.cpp

```diff
@@ -1132,142 +1135,552 @@ void PrestressMembraneElement::CalculateMembraneElasticityTensor(
+        if(rCurrentProcessInfo[TIME] ==1 && mAnisotropicPrestress == true){
+            
+            mG1Initial[PointNumber] = mG1[PointNumber];
+            mG2Initial[PointNumber] = mG2[PointNumber];
+            
+            // out-of-plane direction
+            CrossProduct(mG3Initial[PointNumber],mG1Initial[PointNumber],mG2Initial[PointNumber]);
+            mG3Initial[PointNumber] /= norm_2(mG3Initial[PointNumber]);    
+            
+        }
+
+        // project prestress in the plane spanned by G1 and G2 and transform it in the cartesian CoSy in reference configuration
+        // -- in case of isotropic prestress: no transformation necessary (isotropic state is the same in every (cartesian) CoSy)
+        // -- in case of anisotropic prestress: projection is only executed in the first step, in the following steps the prestress is updated 
+        //    based on the current state
+        if(mAnisotropicPrestress && rCurrentProcessInfo[TIME] == 1){
+            // Initialize Prestress
+            for (unsigned int i_strain=0; i_strain<strain_size; i_strain++){
+                prestress_variable(i_strain,PointNumber) = GetProperties()[MEMBRANE_PRESTRESS](0,i_strain);
+                if(GetProperties().Has(PRESTRESS_AXIS_2_GLOBAL) && GetProperties().Has(PRESTRESS_AXIS_1_GLOBAL) == true){
+                    prestress_axis_1(i_strain,PointNumber) = GetProperties()[PRESTRESS_AXIS_1_GLOBAL](0,i_strain);
+                    prestress_axis_2(i_strain,PointNumber) = GetProperties()[PRESTRESS_AXIS_2_GLOBAL](0,i_strain);
+                }
+            } 
+            // in case that no prestress directions are prescribed: hardcode the prestress direction
+            if (GetProperties().Has(PRESTRESS_AXIS_2_GLOBAL) && GetProperties().Has(PRESTRESS_AXIS_1_GLOBAL) == false){
+                prestress_axis_1(0,PointNumber) = 1;
+                prestress_axis_1(1,PointNumber) = 0;
+                prestress_axis_1(2,PointNumber) = 0;
+            
+                prestress_axis_2(0,PointNumber) = 0;
+                prestress_axis_2(1,PointNumber) = 1;
+                prestress_axis_2(2,PointNumber) = 0;
+            }
+        
+            TransformPrestress(PointNumber);
+        }
+
+        // in case of isotropic prestress: set prestress (no transformation necessary)
+        if(mAnisotropicPrestress == false){
+            for (unsigned int i_strain=0; i_strain<strain_size; i_strain++){
+                prestress_variable(i_strain,PointNumber) = GetProperties()[MEMBRANE_PRESTRESS](0,i_strain);
+            }
+        }
+    }
+
+    // Initialize Material
+
+    if (mConstitutiveLawVector.size() == 0)
+    {
+        mConstitutiveLawVector.resize(integration_points.size());
+
+        for (unsigned int i = 0; i < mConstitutiveLawVector.size(); i++)
+        {
+            mConstitutiveLawVector[i] = GetProperties()[CONSTITUTIVE_LAW]->Clone();
```

Comment:
I am not sure if you can do this.
@RiccardoRossi / @loumalouomega  is this ok?

Reply:
i am a little surprised of this syntax. it depends on how the constitutive law vector is defined (which i don't remember). if it is std::vector< ConstitutiveLaw::Pointer > then it is correct

Reply:
I just copied this part of the code, it was before in the initialize function and now in the initialize solutionstep one. But it work with this code.